pkg/ccl/crosscluster/logical/lww_kv_processor.go

```diff
@@ -321,15 +357,44 @@ func newKVTableWriter(
 	}, nil
 }
 
+// writeableColumns are 'writable' in the sense that they are stored on disk in the primary index.
+//
+// We assume this is all public non-virtual columns and all virtual columns stored in the primary
+// indexes value or key.
+//
+// NOTE: We don't handle virtual columns that are stored in secondary index keys or values here
+// because we assume tables with such columns were disallowed earlier. This function will need to be
+// updated if that restriction is relaxed.
+func writeableColunms(ctx context.Context, td catalog.TableDescriptor) ([]catalog.Column, error) {
+	// TODO(ssd): Validate with SQL foundations that this is correct.
+	keyColumnIDs := td.GetPrimaryIndex().CollectKeyColumnIDs()
+	storedColumnIDs := td.GetPrimaryIndex().CollectPrimaryStoredColumnIDs()
+
+	pubCols := td.PublicColumns()
+	ret := make([]catalog.Column, 0, len(pubCols))
+	for _, c := range pubCols {
+		if c.IsComputed() && !(keyColumnIDs.Contains(c.GetID()) || storedColumnIDs.Contains(c.GetID())) {
```

Comment:
I think all you need to check is `c.isComputed() && c.Isvirtual()`, given rafi's comment over here: https://github.com/stevendanna/cockroach/blob/ssd/cput-with-lww-2/pkg/sql/catalog/tabledesc/logical_replication_helpers.go#L74

Reply:
I updated this a bit, but I'm pretty sure that we still need to check the key column IDs on the primary index.